flow/record/adapter/gcs.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import re
+from fnmatch import fnmatch
+from typing import Iterator, Union
+
+from google.cloud.storage.client import Client
+from google.cloud.storage.fileio import BlobReader, BlobWriter
```

Comment:
This will pull in some major external dependencies. Maybe good to hide it behind a conditional import and add the dependencies as extras?

Reply:
Adapters are lazily loaded and it's not added as a dependency in pyproject.toml, so it should be fine? Although we could do with an extra `flow.record[gcs]` which does add these dependencies.